src/arduino/7_segment_display/7_segment_display.md

```diff
@@ -0,0 +1,217 @@
+
++ 1 Arduino Uno
++ 1 breadboard
++ 2 motstander 1k Ohm (Fargekode: brun-svart-rød-gull)
++ 11 ledninger
++ 1 7-segment display
+
+![utstyr](utstyr.jpg)
+
+# Steg 2: Utforskning {.activity}
+
+Før vi får displayet til å vise tall, er det nyttig å utforske displayet for å
+se hvilke pinner som lyser opp hvilken del av displayet.
+
+## Sjekkliste {.check}
+
++ Motstand fra de to midterste pinnene på displayet.
++ Ledning fra andre enden av motstandene til ground.
++ Ledning fra pin 13 på Arduino til pinnen nederst til høyre på displayet.
+
+![utforskning](utforskning.jpg)
+
+```cpp
+
+int pin = 13;
+
+void setup() {
+    pinMode(pin, OUTPUT);
+}
+
+void loop() {
+    pinMode(pin, HIGH);
+    delay(500);
+    pinMode(pin, LOW);
+    delay(500);
+}
+
+```
+
+Flytt så rundt på ledningen som står i breadboardet for å se hvilke pinner på
+displayet som får hvilken del av segmentet til å blinke.
+
+# Steg 3: Få displayet til å telle oppover {.activity}
+
+Før vi skriver kode for å få displayet til å telle oppover, er vi nødt til å
+lage den fulle kretsen.
+
+![fullkrets](fullkrets.jpg)
+
+## Sjekkliste {.check}
+
++ Motstand fra de to midterste pinnene på displayet.
++ Ledning fra andre enden av motstandene til ground.
++ Koble resten av ledningene som vist på bildet over.
+
+## Skriv kode {.activity}
+
+```cpp
+void blank() {
+    for (int led = 6; led <= 13; ++led) {
+        digitalWrite(led, LOW);
+    }
+}
+
+void tegn_0() {
+    blank();
+    digitalWrite(7, HIGH);
+    digitalWrite(8, HIGH);
+    digitalWrite(9, HIGH);
+    digitalWrite(12, HIGH);
+    digitalWrite(11, HIGH);
+    digitalWrite(10, HIGH);
+}
+
+void tegn_1() {
+    blank();
+    digitalWrite(9, HIGH);
+    digitalWrite(12, HIGH);
+}
+
+void tegn_2() {
+    blank();
+    digitalWrite(8, HIGH);
+    digitalWrite(9, HIGH);
+    digitalWrite(6, HIGH);
+    digitalWrite(10, HIGH);
+    digitalWrite(11, HIGH);
+}
+
+void tegn_3() {
+    blank();
+    digitalWrite(8, HIGH);
+    digitalWrite(9, HIGH);
+    digitalWrite(6, HIGH);
+    digitalWrite(12, HIGH);
+    digitalWrite(11, HIGH);
+}
+
+void setup() {
+    for (int led = 6; led <= 13; ++led) {
+        pinMode(led, OUTPUT);
+    }
+    blank();
+}
+
+void loop() {
+    tegn_0();
+    delay(500);
+    tegn_1();
+    delay(500);
+    tegn_2();
+    delay(500);
+    tegn_3();
+    delay(500);
+}
+```
+
+### Om funksjoner {.protip}
+
+I denne oppgaven har vi introdusert `void blank()`, `void tegn_0()`,
+`void tegn_1()`, `void tegn_2()` og `void tegn_3()`. Disse kaller vi
+_funksjoner_, og er en måte å gruppere kode på. Når vi i `void loop()` sier
+`tegn_0();` gjør vi et _funksjonskall_ til `void tegn_0()`, og koden i den
+funksjonen kjøres.
+
+Hvis du ser på `void setup()` og `void loop()` er dette også funksjoner, hvor
+`void setup()` kalles ved oppstart av programmet, og `void loop()` kalles på
+nytt og på nytt.
+
+### Utfordringer {.challenge}
+
++ Klarer du å få displayet til å telle fra 0 til 9?
```

Comment:
Den vil ikke være lov. Kunne vært lov med templates, men la oss skåne ungene for det enn så lenge. Er definitivt mulig å løse, f. eks med ett int array som alltid er 8 langt og kun inneholder HIGH/LOW verdier.

Sånn sett er kanskje det her en god oppgave å faktisk bygge på for å lære om løkker i en annen oppgave, hvor vi starter fra denne oppgaven og refaktorerer koden til å bruke arrays og løkker.